core/sql/parser/sqlparser.y

```diff
@@ -14684,6 +14759,11 @@ optional_limit_spec : TOK_LIMIT NUMERIC_LITERAL_EXACT_NO_SCALE
 
 dml_statement : dml_query { $$ = $1; }
 
+               | with_clause_list dml_query
```

Comment:
Same comment as a above about where the parser rule for the WITH clause should be.